src/TimeModeling/seismic_preconditioners.jl

```diff
@@ -161,6 +161,29 @@ end
 
 model_topmute(n::Tuple{Int64,Int64}, mute_end::Array{Float32, 2}, length, x) = vec(mute_end) .* vec(x)
 
+
+function model_topmute(n::Tuple{Int64,Int64,Int64}, mute_end::Array{Integer,2}, length::Int64, x_orig)
+    # Model domain topmute for a velocity model of dimensions n = [nx, nz].
+    x = deepcopy(reshape(x_orig,n))
+    x_out = deepcopy(reshape(x_orig,n))
+    for i = 1:n[2]
+        x_out[:,i,:] = reshape(model_topmute((n[1],n[3]),mute_end[:,i],length,x[:,i,:]),n[1],n[3])
+    end
+    return vec(x_out)
+end
```

Comment:
Still intricated and makes copy for each I

```
model_topmute(n::Tuple{Int64,Int64,Int64}, mute_end::Array{Integer,2}, length::Int64, x_orig) = model_topmute((n[1]*n[2], n[3]), vec(mute_end), lengthx_orig)```

Reply:
Thanks for your suggestion. Updated in the latest commit